security/pkg/pki/ra/k8s_ra.go

```diff
@@ -87,6 +114,7 @@ func (r *KubernetesRA) kubernetesSign(csrPEM []byte, caCertFile string, certSign
 }
 
 // Sign takes a PEM-encoded CSR and cert opts, and returns a certificate signed by k8s CA.
+// It returns the leaf certificate only.
```

Comment:
I took a quick look at the tests and I don't think we have a scenario that verifies only the leaf certificate is returned (and not intermediates).

Reply:
Good point - and we may not have a way to know what various providers do. 

It is a big mess - at least for RA and most cases we should have a single Sign that returns the leaf+intermediates - I have no idea why we ended up with 2 functions, but at least in the other code path the Sign is expected to return one. 

Since there is no way for RA to get the intermediates - I suspect only 'full chain' is viable. 

Will try to clean up a bit more this.